opslevel/resource_opslevel_tag.go

```diff
@@ -1,154 +1,207 @@
-// 	if err != nil {
-// 		return err
-// 	}
-
-// 	tags, err := resource.GetTags(client, nil)
-// 	if err != nil {
-// 		return fmt.Errorf("Unable to get tags from '%s' with id '%s'", resourceType, resourceId)
-// 	}
-// 	id := d.Id()
-// 	tag, err := tags.GetTagById(*opslevel.NewID(id))
-// 	if err != nil || tag == nil {
-// 		return fmt.Errorf(
-// 			"Tag '%s' for type %s with id '%s' not found. %s",
-// 			id,
-// 			resource.ResourceType(),
-// 			resource.ResourceId(),
-// 			err,
-// 		)
-// 	}
-
-// 	if err := d.Set("resource_identifier", d.Get("resource_identifier").(string)); err != nil {
-// 		return err
-// 	}
-
-// 	if err := d.Set("resource_type", string(resource.ResourceType())); err != nil {
-// 		return err
-// 	}
-
-// 	if err := d.Set("key", tag.Key); err != nil {
-// 		return err
-// 	}
-
-// 	if err := d.Set("value", tag.Value); err != nil {
-// 		return err
-// 	}
-
-// 	return nil
-// }
-
-// func resourceTagUpdate(d *schema.ResourceData, client *opslevel.Client) error {
-// 	id := d.Id()
-
-// 	input := opslevel.TagUpdateInput{
-// 		Id:    opslevel.ID(id),
-// 		Key:   opslevel.RefOf(d.Get("key").(string)),
-// 		Value: opslevel.RefOf(d.Get("value").(string)),
-// 	}
-// 	if _, err := client.UpdateTag(input); err != nil {
-// 		return err
-// 	}
-
-// 	d.Set("last_updated", timeLastUpdated())
-// 	return resourceTagRead(d, client)
-// }
-
-// func resourceTagDelete(d *schema.ResourceData, client *opslevel.Client) error {
-// 	id := d.Id()
-// 	if err := client.DeleteTag(opslevel.ID(id)); err != nil {
-// 		return err
-// 	}
-// 	d.SetId("")
-// 	return nil
-// }
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+var _ resource.ResourceWithConfigure = &TagResource{}
+
+var _ resource.ResourceWithImportState = &TagResource{}
+
+func NewTagResource() resource.Resource {
+	return &TagResource{}
+}
+
+// TagResource defines the resource implementation.
+type TagResource struct {
+	CommonResourceClient
+}
+
+// TagResourceModel describes the Domain managed resource.
+type TagResourceModel struct {
+	TargetResource types.String `tfsdk:"resource_identifier"`
+	TargetType     types.String `tfsdk:"resource_type"`
+	Key            types.String `tfsdk:"key"`
+	Value          types.String `tfsdk:"value"`
```

Comment:
The Computed `returned_resource` field from [tag docs](https://registry.terraform.io/providers/OpsLevel/opslevel/latest/docs/resources/tag) is missing from this model and the schema